<launch>
    <arg name="dev" default="/dev/ttyACM0"/>
    <arg name="baud" default="115200"/>

    <arg name="camera_topic" default="/usb_cam/image_raw/compressed"/>
    <arg name="tilemap_topic" default="/map_image/tile/compressed"/>
    <arg name="fullmap_topic" default="/map_image/full/compressed"/>
    <arg name="map_output_topic" default="/send_map"/>
    <arg name="jpeg_quality_level" default="40"/>

    <arg name="maestro_topic" default="recv_data_maestro"/>
    <arg name="roboclaw_topic" default="recv_data_roboclaw"/>

    <arg name="enable" default="True"/>

    <node pkg="nordic_node" type="nordic_send.py" name="nordic_send" output="screen">
        <param name="~dev" value="$(arg dev)"/>
        <param name="~baud" value="$(arg baud)"/>
        <param name="~enable" value="$(arg enable)"/>
        <param name="~camera_topic" value="$(arg camera_topic)"/>
        <param name="~tilemap_topic" value="$(arg tilemap_topic)"/>
        <param name="~jpeg_quality_level" value="$(arg jpeg_quality_level)"/>
        <param name="~map_output_topic" value="$(arg map_output_topic)"/>
        <param name="~fullmap_topic" value="$(arg fullmap_topic)"/>
    </node>

    <node pkg="nordic_node" type="nordic_recv.py" name="nordic_recv" output="screen">
        <param name="~dev" value="$(arg dev)"/>
        <param name="~baud" value="$(arg baud)"/>
        <param name="~enable" value="$(arg enable)"/>
        <param name="~maestro_topic" value="$(arg maestro_topic)"/>
        <param name="~roboclaw_topic" value="$(arg roboclaw_topic)"/>
    </node>

</launch>